### PR TITLE
dxValidator: add some tests for events

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets.editors/validator.tests.js [hunk 1]
```diff
@@ -448,6 +450,32 @@ QUnit.module('Events', {
...
+        this.fixture.stubAdapter.getValue.returns(value);
+
+        validator.validate();
+
+        expectedFailedValidationRule.validator.option('onValidated', handlerAfterChange);
+        this.fixture.stubAdapter.getValue.returns(value);
+
+        validator.validate();
+
+        assert.ok(handler.calledOnce, 'Validated handler should be called');
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validator.tests.js]: We already have such assert:
https://github.com/DevExpress/DevExtreme/pull/11369/files#diff-dc05fa7fdaba59cd05c488434aca063dR434

File: testing/tests/DevExpress.ui.widgets.editors/validator.tests.js [hunk 2]
```diff
@@ -417,8 +417,9 @@ QUnit.module('Events', {
     }
 }, () => {
     QUnit.test('Validated event should fire', function(assert) {
-        const value = ''; const name = 'Login'; const expectedFailedValidationRule = { index: 0, type: 'required', isValid: false, message: 'Login is required', validator: {}, value: value }; const handler = sinon.stub();
-
+        const value = ''; const name = 'Login';
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validator.tests.js]:
```suggestion
        const value = '';
        const name = 'Login';
```

File: testing/tests/DevExpress.ui.widgets.editors/validator.tests.js [hunk 3]
```diff
@@ -417,8 +417,9 @@ QUnit.module('Events', {
...
+        const expectedFailedValidationRule = { index: 0, type: 'required', isValid: false, message: 'Login is required', validator: {}, value: value };
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validator.tests.js]:
```suggestion
        const expectedFailedValidationRule = {
            index: 0,
            type: 'required',
            isValid: false,
            message: 'Login is required',
            validator: {},
            value: value 
        };
```

File: testing/tests/DevExpress.ui.widgets.editors/validator.tests.js [hunk 4]
```diff
@@ -448,6 +450,32 @@ QUnit.module('Events', {
         assert.deepEqual(params.brokenRule, expectedFailedValidationRule, 'Failed rules were passed');
     });
 
+    QUnit.test('Validated event should fire correctly after option runtime change', function(assert) {
+        const value = '';
+        const name = 'Login';
+        const expectedFailedValidationRule = { index: 0, type: 'required', isValid: false, message: 'Login is required', validator: {}, value: value };
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validator.tests.js]:
```suggestion
        const expectedFailedValidationRule = {
            index: 0,
            type: 'required',
            isValid: false,
            message: 'Login is required',
            validator: {},
            value: value
        };
```

File: testing/tests/DevExpress.ui.widgets.editors/validator.tests.js [hunk 5]
```diff
@@ -482,6 +510,67 @@ QUnit.module('Events', {
         assert.strictEqual(params.isValid, true, 'isValid was passed');
         assert.strictEqual(params.brokenRule, null, 'Null should be passed as brokenRule ');
     });
+
+    QUnit.module('Subscription by "on" method', () => {
+        QUnit.test('Validated event should fire', function(assert) {
+            const value = '';
+            const name = 'Login';
+            const expectedFailedValidationRule = { index: 0, type: 'required', isValid: false, message: 'Login is required', validator: {}, value: value };
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validator.tests.js]:
```suggestion
            const expectedFailedValidationRule = {
                index: 0,
                type: 'required',
                isValid: false,
                message: 'Login is required',
                validator: {},
                value: value
            };
```

File: testing/tests/DevExpress.ui.widgets.editors/validator.tests.js [hunk 6]
```diff
@@ -482,6 +510,67 @@ QUnit.module('Events', {
...
+                validator: validator,
+                value: value
+            }], 'Correct rules were passed');
+            assert.deepEqual(params.brokenRule, expectedFailedValidationRule, 'Failed rules were passed');
+        });
+
+        QUnit.test('Validated event should fire correctly after option runtime change', function(assert) {
+            const value = '';
+            const name = 'Login';
+            const expectedFailedValidationRule = { index: 0, type: 'required', isValid: false, message: 'Login is required', validator: {}, value: value };
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validator.tests.js]:
```suggestion
            const expectedFailedValidationRule = {
                index: 0,
                type: 'required',
                isValid: false,
                message: 'Login is required',
                validator: {},
                value: value
            };
```

File: testing/tests/DevExpress.ui.widgets.editors/validator.tests.js [hunk 7]
```diff
@@ -482,6 +510,67 @@ QUnit.module('Events', {
...
+                name: name,
+                validationRules: [{ type: 'required' }]
+            });
+
+            expectedFailedValidationRule.validator = validator;
+            this.fixture.stubAdapter.getValue.returns(value);
+
+            validator.on('validated', handler);
+            validator.validate();
+            assert.ok(handler.calledOnce, 'Validated handler should be called');
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validator.tests.js]: We already have this assert here:
https://github.com/DevExpress/DevExtreme/pull/11369/files#diff-dc05fa7fdaba59cd05c488434aca063dR531

File: testing/tests/DevExpress.ui.widgets.editors/validator.tests.js [hunk 8]
```diff
@@ -417,11 +417,20 @@ QUnit.module('Events', {
...
 }, () => {
     QUnit.test('Validated event should fire', function(assert) {
-        const value = ''; const name = 'Login'; const expectedFailedValidationRule = { index: 0, type: 'required', isValid: false, message: 'Login is required', validator: {}, value: value }; const handler = sinon.stub();
-
+        const value = '';
+        const name = 'Login';
+        const expectedFailedValidationRule = {
+            index: 0,
+            type: 'required',
+            isValid: false, message:
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validator.tests.js]:
```suggestion
            isValid: false,
            message: 'Login is required',
```

File: testing/tests/DevExpress.ui.widgets.editors/validator.tests.js [hunk 9]
```diff
@@ -417,11 +417,20 @@ QUnit.module('Events', {
...
+            'Login is required',
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validator.tests.js]:
```suggestion
```

File: testing/tests/DevExpress.ui.widgets.editors/validator.tests.js [hunk 10]
```diff
@@ -417,11 +417,20 @@ QUnit.module('Events', {
...
+            validator: {},
+            value: value
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validator.tests.js]:
```suggestion
            value
```

File: testing/tests/DevExpress.ui.widgets.editors/validator.tests.js [hunk 11]
```diff
@@ -448,6 +458,33 @@ QUnit.module('Events', {
         assert.deepEqual(params.brokenRule, expectedFailedValidationRule, 'Failed rules were passed');
     });
 
+    QUnit.test('Validated event should fire correctly after option runtime change', function(assert) {
+        const value = '';
+        const name = 'Login';
+        const expectedFailedValidationRule = {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validator.tests.js]: Now we do not use this object. Let's remove it?

File: testing/tests/DevExpress.ui.widgets.editors/validator.tests.js [hunk 12]
```diff
@@ -482,6 +524,129 @@ QUnit.module('Events', {
...
+                validator: validator,
+                value: value
+            }], 'Correct rules were passed');
+            assert.deepEqual(params.brokenRule, expectedFailedValidationRule, 'Failed rules were passed');
+        });
+
+        QUnit.test('Validated event should fire correctly after option runtime change', function(assert) {
+            const value = '';
+            const name = 'Login';
+            const expectedFailedValidationRule = {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validator.tests.js]: Now we do not use this object. Let's remove it?

File: testing/tests/DevExpress.ui.widgets.editors/validator.tests.js [hunk 13]
```diff
@@ -482,6 +524,129 @@ QUnit.module('Events', {
...
+    });
+
+    QUnit.module('Subscription by "on" method', () => {
+        QUnit.test('Validated event should fire', function(assert) {
+            const value = '';
+            const name = 'Login';
+            const expectedFailedValidationRule = {
+                index: 0,
+                type: 'required',
+                isValid: false, message:
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validator.tests.js]:
```suggestion
                isValid: false,
                message: 'Login is required',
```

File: testing/tests/DevExpress.ui.widgets.editors/validator.tests.js [hunk 14]
```diff
@@ -482,6 +524,129 @@ QUnit.module('Events', {
...
+                'Login is required',
+                validator: {},
+                value: value
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validator.tests.js]:
```suggestion
                value
```

File: testing/tests/DevExpress.ui.widgets.editors/validator.tests.js [hunk 15]
```diff
@@ -482,6 +524,129 @@ QUnit.module('Events', {
...
+
+    QUnit.module('Subscription by "on" method', () => {
+        QUnit.test('Validated event should fire', function(assert) {
+            const value = '';
+            const name = 'Login';
+            const expectedFailedValidationRule = {
+                index: 0,
+                type: 'required',
+                isValid: false, message:
+                'Login is required',
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validator.tests.js]:
```suggestion
```

File: testing/tests/DevExpress.ui.widgets.editors/validator.tests.js [hunk 16]
```diff
@@ -482,6 +524,129 @@ QUnit.module('Events', {
...
+
+            validator.validate();
+
+            assert.ok(handlerAfterChange.calledOnce, 'Validated handler should be called after option change');
+        });
+
+        QUnit.test('optionChange raising', function(assert) {
+            const optionChangeHandler = sinon.stub();
+            const validator = this.fixture.createValidator({
+                name: 'a',
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validator.tests.js]:
```suggestion
                name: 'a'
```

**Review by San4es — APPROVED** [review]
LGTM

**Review by alexander-kotov-dx — APPROVED** [review]

**Pull request merged**